src/Framework/PaymentGateways/Webhooks/EventHandlers/SubscriptionRenewalDonationCreated.php

```diff
@@ -24,7 +24,13 @@ public function __invoke(
     ) {
         $subscription = give()->subscriptions->getByGatewaySubscriptionId($gatewaySubscriptionId);
 
-        if ( ! $subscription) {
+        if ( ! $subscription || $subscription->initialDonation()->gatewayTransactionId === $gatewayTransactionId) {
+            return;
+        }
```

Comment:
Might be worth separating these circumstances and adding some logging, so customers/support can see this happening.

Reply:
I agree, we should add some logging here

Reply:
I agree, these logs can be really helpful. Resolved in 156a0f280a2f1a45c3403954d3e09a935408fab0 and bd025e3bab29a8bb49124b979568107db2420cf1